iphone/cli/lib/process-js-task.js

```diff
@@ -0,0 +1,374 @@
+'use strict';
+
+const appc = require('node-appc');
+const { IncrementalFileTask } = require('appc-tasks');
+const crypto = require('crypto');
+const fs = require('fs-extra');
+const jsanalyze = require('node-titanium-sdk/lib/jsanalyze');
+const path = require('path');
+
+const i18n = appc.i18n(__dirname);
+const __ = i18n.__;
+
+/**
+ * Task that processes JS files by applying several transforms and copying them to their
+ * destination inside the build directory.
+ */
+class ProcessJsTask extends IncrementalFileTask {
+	/**
+	 * Constructs a new processing task.
+	 *
+	 * @param {Object} taskInfo Configuration object for this task
+	 */
+	constructor(taskInfo) {
+		taskInfo.name = taskInfo.name || 'process-js';
+		super(taskInfo);
+
+		this.jsFiles = taskInfo.jsFiles;
+		this.jsBootstrapFiles = taskInfo.jsBootstrapFiles;
+		this.builder = taskInfo.builder;
+		this.sdkCOmmonFolder = taskInfo.sdkCOmmonFolder;
+
+		const minify = this.builder.minifyJS;
+		const transpile = this.builder.transpile;
+		this.defaultAnalyzeOptions = {
```

Comment:
feels like this object should just get passed in whole, rather than re-generated from various values pulled from the builder